File: CapaEntidades/Paciente.cs
using System;

namespace CapaEntidades
{
    public class Paciente
    {
        public int IdPaciente { get; set; }
        public String Nombres { get; set; }
        public String ApPaterno { get; set; }
        public String ApMaterno { get; set; }
        public int Edad { get; set; }
        public char Sexo { get; set; }
        public String NroDocumento { get; set; }
        public String Direccion { get; set; }
        public String Telefono { get; set; }
        public bool Estado { get; set; }
        public String Imagen { get; set; }

        public Paciente()
        {
        }

        public Paciente(int IdPaciente, string Nombres, string ApPaterno, string ApMaterno, int Edad, char Sexo, string NroDocumento, string Direccion, string Telefono, bool Estado, string Imagen)
        {
            this.IdPaciente = IdPaciente;
            this.Nombres = Nombres;
            this.ApPaterno = ApPaterno;
            this.ApMaterno = ApMaterno;
            this.Edad = Edad;
            this.Sexo = Sexo;
            this.NroDocumento = NroDocumento;
            this.Direccion = Direccion;
            this.Telefono = Telefono;
            this.Estado = Estado;
            this.Imagen = Imagen;
        }
    }
}

File: CapaPresentacion/js/paciente.js
//function templateRow() {
//    var template = "<tr>";
//    template += ("<td>" + "123" + "</td>");
//    template += ("<td>" + "123" + "</td>");
//    template += ("<td>" + "123" + "</td>");
//    template += ("<td>" + "123" + "</td>");
//    template += ("<td>" + "123" + "</td>");
//    template += ("<td>" + "123" + "</td>");
//    template += "</tr>"
//    return template;
//}

//function addRowToDataTable() {
//    var template = templateRow();
//    for (var i = 0; i < 10; i++) {
//        $("#tbl_body-table").append(template);
//    }
//}

//addRowToDataTable();



function addRowDT(data) {
    var tabla = $("#tbl_pacientes").DataTable();
    for (var i = 0; i < data.length; i++) {
        tabla.fnAddData([
            data[i].IdPaciente,
            data[i].Nombres,
            (data[i].ApPaterno + " " + data[i].ApMaterno),
            ((data[i].Sexo == 'M')? "Masculino": "Femenino"),
            data[i].Edad,
            data[i].Direccion,
            ((data[i].Estado == true)? "Activo": "Inactivo"),
        ]);
    }
}

//Llamamos la funcion mediante ajax
function sendDataAjax() {
    $.ajax({
        type: "POST",
        url: "GestionarPaciente.aspx/ListarPacientes",
        data: {},
        contentType: 'application/json; charset=utf-8',
        error: function (xhr, ajaxOptions, thrownError) {
            console.log(xhr.status + " \n" + xhr.responseText, "\n" + thrownError);
        },
        success: function (data) {
            //console.log(data);
            addRowDT(data.d);
        }
    });
}

//Llamando a la funcion de ajax al cargar el documento
sendDataAjax();